src/bin/06_storage_iter.rs

```diff
@@ -0,0 +1,122 @@
+
+    // Hexify the above bytes and make the request to get the value back:
+    let storage_key_hex = format!("0x{}", hex::encode(&storage_key));
+    println!("VersionNotifiers storage key: {}", storage_key_hex);
+
+    let params = rpc_params![storage_key_hex.clone()];
+    let keys: Vec<StorageKey> = client.request("state_getKeys", params).await?;
+
+    println!("Obtained keys:");
+    for key in keys.iter() {
+        let key_hex = format!("0x{}", hex::encode(&key));
+        println!("Key: {}", key_hex);
+
+        // Obtain the byte representation of the key.
+        let key_bytes = &key.0;
+        // The first 32 bytes are identical to our storage query key and are represented by:
+        // The first 16 bytes of the key represent the `storage_prefix`.
+        let inspected_bytes = &key_bytes[0..16];
+        println!(
+            "     bytes[ 0..16]: {} == twox_128(\"XcmPallet\")",
+            format!("0x{}", hex::encode(&inspected_bytes))
+        );
+        // The next 16 bytes represent the `storage_name_hashed`.
+        let inspected_bytes = &key_bytes[16..32];
+        println!(
+            "     bytes[16..32]: {} == twox_128(\"VersionNotifiers\")",
+            format!("0x{}", hex::encode(&inspected_bytes))
+        );
+
+        // The next 8 bytes are represented by the twox64(first key), the hashing produces 8 bytes.
+        let inspected_bytes = &key_bytes[32..40];
+        println!(
+            "     bytes[32..40]: {}                 == twox_64(first key)",
+            format!("0x{}", hex::encode(&inspected_bytes))
+        );
+
+        // The next 4 bytes represent the first key - `xcm::Version` which is an u32.
+        let inspected_bytes = &key_bytes[40..44];
+        let xcm_version = xcm::Version::decode(&mut inspected_bytes.clone())?;
+        println!(
+            "     bytes[40..44]: {}                         == first key - XcmVersion: {}",
+            format!("0x{}", hex::encode(&inspected_bytes)),
+            xcm_version
+        );
+
+        // The next 16 bytes represent the blake2_128(second key), the hashing produces 16 bytes.
+        let inspected_bytes = &key_bytes[44..60];
+        println!(
+            "     bytes[44..60]: {} == blake2_128(second key)",
+            format!("0x{}", hex::encode(&inspected_bytes))
+        );
+
+        // The remaining bytes are represented by the second key.
+        let inspected_bytes = &key_bytes[60..];
+        let versioned_multilocation =
+            xcm::VersionedMultiLocation::decode(&mut inspected_bytes.clone())?;
+        println!(
+            "     bytes[60..  ]: {}                     == second key - VersionedMultiLocation: {:?}",
+            format!("0x{}", hex::encode(&inspected_bytes)),
+            versioned_multilocation
+        );
+
+        // Get the value of the storage key.
+        let params = rpc_params![key];
+        let result_hex: String = client.request("state_getStorage", params).await?;
```

Comment:
Is there a need to use a WS client and not make an http request instead? (I don't mind too much, but trying to keep these examples pretty barebones so unless we need to use jsonrpsee I'd probably rather keep it low level with the rpc_to_localhost call just so that people can see exactly what's being sent over the wire :))

Reply:
Have added a comment wrt the WS client. For this example, the WS client makes RPC calls to the localhost node.
However, the same example can be modified toward a production node 😄 Which could provide a bit more data (ie. the localhost node must be synced in order to provide any meaningful double storage maps)